src/Icon.php

```diff
@@ -0,0 +1,136 @@
+<?php
+/**
+ * Class Icon.
+ *
+ * @package AmpProject\AmpWP
+ */
+
+namespace AmpProject\AmpWP;
+
+/**
+ * Icons used to visually represent the state of a validation error.
+ *
+ * @package AmpProject\AmpWP
+ */
+class Icon {
+
+	/**
+	 * Indicates there are validation errors for the AMP page.
+	 */
+	const INVALID = 'amp-invalid';
+
+	/**
+	 * Indicate an AMP version of the page is available.
+	 */
+	const LINK = 'amp-link';
+
+	/**
+	 * Indicates the page is valid AMP.
+	 */
+	const VALID = 'amp-valid';
+
+	/**
+	 * Indicates there are validation errors which have not been explicitly accepted.
+	 */
+	const WARNING = 'amp-warning';
+
+	/**
+	 * Current icon.
+	 *
+	 * @var string
+	 */
+	private static $icon;
```

Comment:
No need for this to be static. You have named constructors below, so you are always instantiating an actual object instance.

Reply:
Done via https://github.com/ampproject/amp-wp/pull/4382/commits/04eacacab8ff40899eeb41c79f3801137882afb5.